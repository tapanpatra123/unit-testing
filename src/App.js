import React, { PureComponent } from "react";
import Counter from './components/counter'
import Footer from './components/footer'
import Header from './components/header'

export default class App extends PureComponent {

  render() {
    return (
      <div className="app">
        <Header/>
        <Counter name="Title Goes Here"/>
        <Footer name={"footer contenet goes here"}/>
      </div>
    );
  }
}

