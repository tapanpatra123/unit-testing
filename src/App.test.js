import React from "react";
import {shallow} from 'enzyme';
import { shallowToJson} from 'enzyme-to-json';
import App from './App';
import Header from './components/header'
import Counter from './components/counter'
import Footer from './components/footer'

describe("App Component",()=>{

    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<App/>);
    })

    it("to snapshot",()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })

    it("should render Header component", ()=>{
        const header = wrapper.find(Header);
        expect(header.exists()).toBe(true)
    })

    it("should render Counter component", ()=>{
        const counter = wrapper.find(Counter);
        expect(counter.exists()).toBe(true)
    })

    it("should render counter component", ()=>{
        const counter = wrapper.find(Counter);
        expect(counter.exists()).toBe(true)
    })

    it("should render footer component", ()=>{
        const counter = wrapper.find(Footer);
        expect(counter.exists()).toBe(true)
    })
})