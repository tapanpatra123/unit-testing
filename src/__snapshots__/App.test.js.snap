// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component to snapshot 1`] = `
<div
  className="app"
>
  <Header />
  <Counter
    name="Title Goes Here"
  />
  <Footer
    name="footer contenet goes here"
  />
</div>
`;
